marlowe-playground-client/src/MarloweEditor/BottomPanel.purs

```diff
@@ -1,37 +1,31 @@
-module Simulation.BottomPanel where
+module MarloweEditor.BottomPanel (bottomPanel) where
 
 import Control.Alternative (map)
-import Data.Array (concatMap, drop, head, length, reverse)
+import Data.Array (drop, head, length)
 import Data.Array as Array
-import Data.BigInteger (BigInteger)
-import Data.Either (Either(..), either)
 import Data.Eq (eq, (==))
 import Data.Foldable (foldMap)
-import Data.HeytingAlgebra (not, (||))
-import Data.Lens (_Just, has, only, previewOn, to, (^.))
-import Data.Lens.NonEmptyList (_Head)
+import Data.HeytingAlgebra (not)
+import Data.Lens (to, (^.))
 import Data.List (List, null, toUnfoldable)
 import Data.List as List
 import Data.List.NonEmpty (toList)
-import Data.Map as Map
-import Data.Maybe (Maybe(..), isJust, isNothing)
-import Data.Newtype (unwrap)
+import Data.Maybe (Maybe(..))
 import Data.String (take)
 import Data.String.Extra (unlines)
-import Data.Tuple (Tuple(..))
 import Data.Tuple.Nested ((/\))
-import Halogen.Classes (aHorizontal, accentBorderBottom, active, activeClass, closeDrawerArrowIcon, collapsed, first, flex, flexLeft, flexTen, footerPanelBg, minimizeIcon, rTable, rTable4cols, rTableCell, rTableDataRow, rTableEmptyRow, spanText, underline)
+import Halogen.Classes (aHorizontal, accentBorderBottom, active, activeClass, closeDrawerArrowIcon, collapsed, flex, flexLeft, flexTen, footerPanelBg, minimizeIcon, spanText, underline)
 import Halogen.Classes as Classes
-import Halogen.HTML (ClassName(..), HTML, a, a_, b_, br_, button, div, h2, h3, img, li, li_, ol, pre, section, span_, strong_, text, ul, ul_)
+import Halogen.HTML (ClassName(..), HTML, a, a_, b_, br_, button, div, h2, h3, img, li, li_, ol, pre, section, span_, text, ul, ul_)
 import Halogen.HTML.Events (onClick)
 import Halogen.HTML.Properties (alt, class_, classes, enabled, src)
-import Marlowe.Semantics (Assets(..), ChoiceId(..), Input(..), Party, Payee(..), Payment(..), Slot(..), SlotInterval(..), Token(..), TransactionInput(..), TransactionWarning(..), ValueId(..), _accounts, _boundValues, _choices, timeouts)
+import Marlowe.Semantics (ChoiceId(..), Input(..), Payee(..), Slot(..), SlotInterval(..), TransactionInput(..), TransactionWarning(..))
 import Marlowe.Symbolic.Types.Response as R
+import MarloweEditor.Types (Action(..), AnalysisState(..), BottomPanelView(..), MultiStageAnalysisData(..), State, _analysisState, _bottomPanelView, _editorErrors, _editorWarnings, _showBottomPanel, _showErrorDetail, isValidContract)
 import Network.RemoteData (RemoteData(..), isLoading)
-import Prelude (bind, const, mempty, pure, show, zero, ($), (&&), (<$>), (<<<), (<>))
-import Pretty (renderPrettyParty, renderPrettyPayee, renderPrettyToken, showPrettyMoney)
+import Pretty (showPrettyToken)
+import Prelude (bind, const, mempty, pure, show, ($), (&&), (<$>), (<<<), (<>))
```

Comment:
while we are here, I think it might be better for the Prelude to use `hiding` instead of explicitly importing these things (only for `Prelude`). It's probably `import Prelude hiding (div)`

Reply:
Yeah, I don't like having to specify imports of the Prelude, so I went ahead and changed it in all marlowe-client files 😅